src/InstallerHosts/Elastic.InstallerHosts/Elasticsearch/Tasks/DeleteDirectoriesTask.cs

```diff
@@ -27,20 +27,28 @@ public DeleteDirectoriesTask(ElasticsearchInstallationModel model, ISession sess
 		protected override bool ExecuteTask()
 		{
 			if (this.InstallationModel.NoticeModel.AlreadyInstalled && !this.Session.Uninstalling)
+			{
+				this.Session.Log($"Skipping {nameof(DeleteDirectoriesTask)}: previously installed and not currently uninstalling");
+				return true;
+			}
+			
+			if (this.InstallationModel.NoticeModel.AlreadyInstalled && this.RollBack)
```

Comment:
I think this can be simplified with the above `if` statement

```csharp
if (this.InstallationModel.NoticeModel.AlreadyInstalled)
{
    if (!this.Session.Uninstalling)
    {
        this.Session.Log($"Skipping {nameof(DeleteDirectoriesTask)}: previously installed and not currently uninstalling");
        return true;
    }
    else if (this.RollBack)
    {
        this.Session.Log($"Skipping {nameof(DeleteDirectoriesTask)}: previously installed and currently performing a rollback");
        return true;
    }
}
```